src/modules/mc_pos_control/PositionControl/ControlMath.hpp

```diff
@@ -51,7 +51,8 @@ namespace ControlMath
  * @param yaw_sp the desired yaw
  * @return vehicle_attitude_setpoints_s structure
  */
-vehicle_attitude_setpoint_s thrustToAttitude(const matrix::Vector3f &thr_sp, const float yaw_sp);
+void thrustToAttitude(vehicle_attitude_setpoint_s &attitude_setpoint, const matrix::Vector3f &thr_sp,
```

Comment:
I'd turn it around: the convention is input params, then output params:
void thrustToAttitude(const matrix::Vector3f &thr_sp, const float yaw_sp, vehicle_attitude_setpoint_s &attitude_setpoint);

Reply:
I can totally do that I just didn't know where this convention is.
I was following the http://www.cplusplus.com/reference/cstring/memcpy/ convention which might not be applicable.

Reply:
Maybe this is worth taking an example from: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-out-multi :+1:

Reply:
I'm convinced, fixed this in https://github.com/PX4/Firmware/pull/13381/commits/3d845dea9d31c4f4ab7d2989bb48f1e60698922d and will follow the convention from now on.